nucleus/payara-modules/opentracing-adapter/src/main/java/fish/payara/opentracing/OpenTracingService.java

```diff
@@ -154,15 +171,20 @@ public synchronized Tracer getTracer(String applicationName) {
      * @return True if the Request Tracing Service is enabled
      */
     public boolean isEnabled() {
-        RequestTracingService requestTracingService = getFromServiceHandle(Globals.getDefaultBaseServiceLocator(),
-                RequestTracingService.class);
-
-        if (requestTracingService != null) {
+        if (getRequestTracingService() != null) {
             return requestTracingService.isRequestTracingEnabled();
         }
         return false;
     }
 
+    public RequestTracingService getRequestTracingService() {
```

Comment:
How about returning an Optional<RequestTracingService>?

Reply:
I don't think returning an Optional really adds much here.
The method is intended as a poke to HK2 if required rather than any kind of true getter (public scope currently present is wrong - should be private).